test/e2e/create_staticip_test.go

```diff
@@ -74,11 +73,6 @@ var _ = Describe("Podman create with --ip flag", func() {
 		// test1 container is running with the given IP.
 		result = podmanTest.Podman([]string{"start", "-a", "test2"})
 		result.WaitWithDefaultTimeout()
-		Expect(result).To(ExitWithError())
-		if podmanTest.NetworkBackend == CNI {
-			Expect(result.ErrorToString()).To(ContainSubstring("requested IP address %s is not available", ip))
-		} else if podmanTest.NetworkBackend == Netavark {
-			Expect(result.ErrorToString()).To(ContainSubstring("requested ip address %s is already allocated", ip))
-		}
+		Expect(result).To(ExitWithError(125, " is already allocated"))
```

Comment:
@Luap99 is this what you meant?

Reply:
Fair enough. I think `ExitWithError(xxx, fmt.Sprintf()`)` looks clunky, but tighter testing is better testing. So done.

`ContainSubstring()` is kind of nice in how it takes varargs. I think that's nicely readable, but I just didn't want to complicate this PR even more by adding that to `ExitWithError()`. Would that be worth pursuing once this exercise is done and I tighten up the `ExitWithError()` signature?

Reply:
> ContainSubstring() is kind of nice in how it takes varargs. I think that's nicely readable, but I just didn't want to complicate this PR even more by adding that to ExitWithError(). Would that be worth pursuing once this exercise is done and I tighten up the ExitWithError() signature?

Yes I think that would be valuable but definitely not worth to block this work over it.